### PR TITLE
Enable stricter validation for build-logic convention plugins

### PR DESCRIPTION
**Pull request by alexvanyo**
Enables stricter validation for the convention plugins, and fixes the resulting reported issues around caching and file sensitivity.

### CODE REVIEW

File: .github/workflows/Build.yaml [hunk 1]
```diff
@@ -39,6 +39,9 @@ jobs:
       - name: Setup Gradle
         uses: gradle/gradle-build-action@v2
 
+      - name: Check build-logic
+        run: ./gradlew check -p build-logic
```
[the remark this answers is not in the document]
Comment (SimonMarquis) [reply .github/workflows/Build.yaml:43]: Although I guess it should even be more like `./gradlew :build-logic:convention:check`

File: build-logic/convention/build.gradle.kts [hunk 2]
```diff
@@ -43,6 +43,13 @@ dependencies {
     compileOnly(libs.ksp.gradlePlugin)
 }
 
+tasks {
+    validatePlugins {
+        enableStricterValidation.set(true)
+        failOnWarning.set(true)
```
[the remark this answers is not in the document]
Comment (alexvanyo) [reply build-logic/convention/build.gradle.kts]: Updated

File: .github/workflows/Build.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (alexvanyo) [reply .github/workflows/Build.yaml:43]: Right, `./gradlew :build-logic:check` doesn't work, but `./gradlew :build-logic:convention:check` does work - although it does look to be different in a subtle way.

When I run `./gradlew :build-logic:convention:check`, it has to configure the rest of the project modules, but if I run `./gradlew check -p build-logic, it seems to ignore the rest of the project, saving time.

I can't tell why there's a difference though.

**Review by javadude — APPROVED** [review]

File: .github/workflows/Build.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (javadude) [reply .github/workflows/Build.yaml:43]: Is there anything that needs to be done based on this conversation?

[the remark this answers is not in the document]
Comment (alexvanyo) [reply .github/workflows/Build.yaml:43]: I don't believe so, resolving.

**Review by javadude — APPROVED** [review]

**Pull request merged**